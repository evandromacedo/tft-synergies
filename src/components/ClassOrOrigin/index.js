import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import capitalize from 'lodash/capitalize';
import SynergyIcon from '../SynergyIcon';

const Grid = styled.div`
  display: grid;
  grid-gap: 0.6rem;
  grid-template-columns: 2rem auto;
  align-items: center;
`;

const Title = styled.p`
  ${props => props.theme.bold12px}
  text-shadow: 0px 0px 6px black;

  /* Necessary to align the text on the center due to font issues. */
  padding-top: 2px;
`;

function ClassOrOrigin({ type }) {
  return (
    <Grid>
      <SynergyIcon classOrOrigin={type} type="default" style={{ fontSize: '2rem' }} />
      <Title>{capitalize(type)}</Title>
    </Grid>
  );
}

ClassOrOrigin.propTypes = {
  type: PropTypes.oneOf([
    'demon',
    'dragon',
    'exile',
    'glacial',
    'imperial',
    'noble',
    'ninja',
    'pirate',
    'phantom',
    'robot',
    'void',
    'wild',
    'yordle',
    'assassin',
    'blademaster',
    'brawler',
    'elementalist',
    'guardian',
    'gunslinger',
    'knight',
    'ranger',
    'shapeshifter',
    'sorcerer',
    'hextech'
  ]).isRequired
};

export default ClassOrOrigin;
