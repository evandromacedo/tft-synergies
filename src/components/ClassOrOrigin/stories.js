import React from 'react';
import ClassOrOrigin from './';
import { storiesOf } from '@storybook/react';

storiesOf('ClassOrOrigin', module)
  .add('Demon', () => <ClassOrOrigin type="Demon" />)
  .add('Dragon', () => <ClassOrOrigin type="Dragon" />);
