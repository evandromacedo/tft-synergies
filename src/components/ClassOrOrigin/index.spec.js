import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ClassOrOrigin from '.';

describe('<ClassOrOrigin />', () => {
  it('render properly and matches snapshot', () => {
    const wrapper = shallow(<ClassOrOrigin type="demon" />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
