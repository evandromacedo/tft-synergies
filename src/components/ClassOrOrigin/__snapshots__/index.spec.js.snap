// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClassOrOrigin /> render properly and matches snapshot 1`] = `
<styled.div>
  <SynergyIcon
    classOrOrigin="demon"
    type="default"
  />
  <styled.p>
    Demon
  </styled.p>
</styled.div>
`;
